locust/dispatch.py

```diff
@@ -301,7 +304,9 @@ def _get_user_current_count(self, user: str) -> int:
 
     def _distribute_users(
         self, target_user_count: int
-    ) -> Tuple[dict, Generator[str, None, None], typing.Iterator["WorkerNode"], List[Tuple["WorkerNode", str]]]:
+    ) -> Tuple[
+        Dict[str, Dict[str, int]], Generator[Optional[str], None, None], itertools.cycle, List[Tuple["WorkerNode", str]]
```

Comment:
Is `str` really optional? Feels like a different solution could be possible?

Reply:
it was due to:
```
def infinite_cycle_gen(users: List[Tuple[Type[User], int]]) -> itertools.cycle:
            if not users:
                return itertools.cycle([None])
```

i made quick check, and change it to `itertools.cycle([])` and removed `Optional` and `mypy` didn't complain. just about to logout, but i'll run the tests tomorrow to make sure they still pass as well.

Reply:
This one was a bit tricky, `itertools.cycle([None])` gives you an endless loop of `None` values, and where these values are used there's checks, like:
```
        while user_count < target_user_count:
            user = next(user_gen)
            if not user:
                break
```

and,
```
        for user in self._user_generator:
            if not user:
                self._no_user_to_spawn = True
                break
```

While `itertools.cycle([])` gives you nothing, e.g., it would be like `for _ in []`.

So I would say that yes, it is an `Optional[str]`, otherwise there would have to be quite some logic thas has to be re-written in `dispatch.py`

Reply:
Ok, so this one can be typed as `str` when using `--no-strict-optional` settings for mypy. I'd like to see it typed as `Optional[str]` though :) But your call.